src/LayerEditor/ui/panels/regions_panel.py

```diff
@@ -59,87 +50,85 @@ def select_regions(self, region_dict):
         :return:
         """
         # assert layer_id in self._selected_regions
+        assert False, "Not refactored"
         self._selected_regions.update(region_dict)
         self.region_changed.emit()
 
     def select_layer(self, layer_id):
         """ Select actual layer tab."""
-        self._selected_layers[self.current_topology_id] = layer_id
+        assert False, "Not refactored"
+        self.block.gui_selected_layers[self.current_topology_id] = layer_id
 
-    def add_region(self, name, dim, color):
+    def add_region(self, name, dim):
         """ Add new region according to the provided data from the current tab and select it."""
-        region = self.regions.add_new_region(color, name, dim, True, "Add Region")
-        self._selected_regions[self.current_layer_id] = region.reg_id
+        reg = self.regions_model.add_region(name, dim)
+        self.block.gui_selected_layer.gui_selected_region = reg
+        self.block.set_region_to_selected_shapes(reg)
         self.region_list_changed.emit()
-        return region
+        self.region_changed.emit()
```

Comment:
Possibly move update into BlockModel and emit the signal there.

Reply:
The whole method was moved to LEModel and so was the signal.

Reply:
https://github.com/GeoMop/GeoMop/blob/df105ad9c7689fa3f2b141a522ec791c1d7f7df6/src/LayerEditor/ui/data/le_model.py#L101-L108